lib/matplotlib/widgets.py

```diff
@@ -1072,9 +1071,18 @@ def set_active(self, index):
         if index not in range(len(self.labels)):
             raise ValueError(f'Invalid CheckButton index: {index}')
 
-        l1, l2 = self.lines[index]
-        l1.set_visible(not l1.get_visible())
-        l2.set_visible(not l2.get_visible())
+        if colors.same_color(
+                self._crosses.get_facecolor()[index], colors.to_rgba("none")
+        ):
+            self._crosses.get_facecolor()[index] = colors.to_rgba("k")
+        else:
+            self._crosses.get_facecolor()[index] = colors.to_rgba("none")
```

Comment:
The point made in #24479 is that editing the output of the getter function is not guaranteed to _keep_ those changes (it works, sometimes, because it is not a copy that is returned, but there are other mechanisms at play)

more properly this would be:

```python
facecolors = self._crosses.get_faceolor()
facecolors[index] = colors.to_rgba("k") # or "none"
self._crosses.set_facecolor(facecolors)
```

This calls the setter which updates the internal state properly

(This also does need to be corrected in the radio button implementation)